hosting_docs/source/installation/prereqs/2-architecture.rst

```diff
@@ -1,4 +1,284 @@
\ No newline at end of file
+.. image:: img/architecture_01.png
+
+Primary Databases
+-----------------
+
++------------------------------+-----------------------------+
+| CouchDB (aka "couch")        | PostgreSQL (aka "postgres") |
++==============================+=============================+
+| JSON documents               | Table rows                  |
++------------------------------+-----------------------------+
+| Schemaless                   | Schema                      |
++------------------------------+-----------------------------+
+| Map/reduce: Go through every | Dynamic queries             |
+| doc in the table to create   |                             |
+| an index. Run the query on   |                             |
+| the indexed data.            |                             |
++------------------------------+-----------------------------+
+| Not relational: Can't join   | Relational queries          |
+| across tables.               |                             |
++------------------------------+-----------------------------+
+
+We started with Couch. We struggled with some aspects of the way it
+works, and so we added Postgres. All forms and cases are stored in
+Postgres now.
+
+A Couch document is analogous to a SQL row. This is the thing that the
+"doc search" URL returns.
+
+
+What's stored where?
+--------------------
+
+================   ===================   ========
+PostgreSQL         CouchDB               Both
+================   ===================   ========
+Form Submissions   Users                 Products
+Cases              Applications
+Locations          Groups
+UCR data           UCR configs
+SMS models         Exports configs
+Ledgers            Random config
+Random config
+
+UCRs are User-Configurable Reports; reports configured using
+JSON-formatted definitions.
+
+Raw data (forms and cases) → apply "data source config" → Data source →
+apply "report config" → report output. e.g. A project may have a dozen
+reports all based on the same data source. A UCR is a data source config +
+a report config.
+
+Data sources are stored in SQL, but report outputs are run in real time.
+
+Things in the "Both" column are low priority to migrate, and for now are
+going to live on both.
+
+There has been a long period of Couch-to-SQL migration.
+
+"Random config": We like SQL for this more, but a lot of this has been
+on Couch historically, and we're going to leave it there for the time
+being. There is no logical grouping of configs on SQL vs. Couch. About
+300-400 documents in Couch (each one of which would need to be migrated
+individually). About 200 documents in SQL. "Random config" is per
+document. e.g.
+
+* "create a new domain" is a request that creates a "domain" document
+  type (in Couch).
+* "invite new web user" is a request that creates a "web user" document
+  type.
+
+
+BlobDB
+------
+
+Riak is a key-value store. BlobDB is the wrapper we've written around
```

Comment:
We haven't used Riak for quite some time. I think the recommended backend is now [Minio](https://docs.min.io/). References should be changed below as well.